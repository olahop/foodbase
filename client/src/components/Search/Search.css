.search {
    width: 100%;
    position: relative;
    display: flex;
    border: 1.5px solid #00B4CC;
    border-radius: 5px;
}

.search:hover, .search:focus-within {
    border: 3px solid #00B4CC;
}

.searchTerm {
    width: 100%;
    padding: 6px;
    height: 24px;
    border:none;
    border-radius: 2px 0 0 2px;
    outline: none;
    color: #9DBFAF;
}

.searchTerm:focus {
    color: black;
}

.searchButton {
    width: 40px;
    height: 36px;
    border: none;
    background: #fff;
    text-align: center;
    color: grey;
    border-radius: 0 2px 2px 0;
    font-size: 15px;
}

.wrap {
    width: 40%;
    min-width: 300px;;
    margin:auto;
}
