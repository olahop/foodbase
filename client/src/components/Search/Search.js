import React from "react";
import { connect } from "react-redux";
import { changeSearch } from "../../actions/search";
import "./Search.css";


export class Search extends React.Component {

    constructor(props){
        super(props)

        this.onChange = this.onChange.bind(this)
    }

    // updates the search value in the redux store when the search field is updated
    onChange = (event) => {
        this.props.onChange(event.target.value)
    }

    render() {
        return (
            <>
              <div className="wrap">
                <div className="search" >
                    <input className="searchTerm" onChange={this.onChange} placeholder="SÃ¸k etter matvare..." />
                    <button type="submit" className="searchButton">
                        <i className="fa fa-search"></i>
                    </button>
                </div> 
              </div>
            </>
        );
    }
}

const mapStateToProps = state => ({
    search: state.search,
});

const mapActionsToProps = {
    onChange: changeSearch
};

export default connect(mapStateToProps,mapActionsToProps)(Search)
