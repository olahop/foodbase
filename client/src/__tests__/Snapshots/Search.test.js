import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import Search from '../../components/Search/Search';
const mockStore = configureStore([]);
describe('Test Search component', () => {
  let store;
  let component;
  
  beforeAll(() => {
    store = mockStore();
    component = renderer.create(
      <Provider store={store}>
        <Search />
      </Provider>
    );
  });

  it('Component renders correctly', () => {
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});