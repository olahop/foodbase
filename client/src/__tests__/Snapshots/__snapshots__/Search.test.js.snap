// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Search component Component renders correctly 1`] = `
<div
  className="wrap"
>
  <div
    className="search"
  >
    <input
      className="searchTerm"
      onChange={[Function]}
      placeholder="SÃ¸k etter matvare..."
    />
    <button
      className="searchButton"
      type="submit"
    >
      <i
        className="fa fa-search"
      />
    </button>
  </div>
</div>
`;
